src/numeric/impl_numeric.rs

```diff
@@ -115,6 +116,42 @@ impl<A, S, D> ArrayBase<S, D>
         sum / &aview0(&cnt)
     }
 
+    /// Return variance along `axis`.
+    ///
+    /// The variance is computed using the Welford one-pass algorithm
+    /// https://www.jstor.org/stable/1266577
+    ///
+    /// ```
+    /// use ndarray::{aview1, arr2, Axis};
+    ///
+    /// let a = arr2(&[[1., 2.],
+    ///                [3., 4.]]);
+    /// let var = a.var_axis(Axis(0));
+    /// println!("{:?}", var);
+    /// assert!(
+    ///     var == aview1(&[1., 1.])
+    /// );
+    /// ```
+    pub fn var_axis(&self, axis: Axis) -> Array<A, D::Smaller>
+        where A: LinalgScalar + ScalarOperand,
```

Comment:
1. ScalarOperand is not intended to be used like this, so I'd try to find a different way to do it, without that trait.
2. Variance usually has a [ddof](https://docs.scipy.org/doc/numpy-1.14.0/reference/generated/numpy.var.html) parameter, and I think we need to allow for it in some way for var and std.
3. We need to find a different way to compute this, to avoid using `.to_owned()` on every row.

Reply:
What do you mean by _ScalarOperand is not intended used like this_?

Reply:
It's a crutch specifically for some operator impls for arrays. Not all traits have the same role.